### PR TITLE
DOCSUP-5823: Documented Map(key,value) type and map() function.

### PR DESCRIPTION
**Pull request by olgarev**
Data type description template also added.

I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- Documentation for #15806

### CODE REVIEW

**Comment by olgarev** [comment]
@CurtizJ Hi, will you review this, pls?

**Comment by olgarev** [comment]
@CurtizJ I had to delete symbolic links because of the errors, now all checks have passed.

**Comment by olgarev** [comment]
@Akazz Hi, can you review these docs, pls?

File: docs/en/sql-reference/data-types/map.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+To get the value from an `a Map('key', 'value')` column, use `a['key']` syntax.
+
+**Example**
+
+Query:
+
+``` sql
+CREATE TABLE table_map (a Map(String, UInt64)) ENGINE=Memory;
+INSERT INTO table_map VALUES ({'key1':1, 'key2':100}), ({'key1':2,'key2':200}), ({'key1':3,'key2':300});
```
Comment (vitlibar) [inline docs/en/sql-reference/data-types/map.md:25]: I think it's better to add more examples where an inserted value doesn't have `key1` or doesn't have keys at all (`{}`).

What happens if the same key is used multiple times, like `{'key1':1000, 'key1':2000}`.

The search for key in map is case sensitive, isn't it?

File: docs/en/sql-reference/data-types/map.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+`Map(key, value)` data type stores `key:value` pairs in structures like JSON. 
+
+**Parameters** 
+-   `key` — The key part of the pair. [String](../../sql-reference/data-types/string.md) or [Integer](../../sql-reference/data-types/int-uint.md).
+-   `value` — The value part of the pair. [String](../../sql-reference/data-types/string.md), [Integer](../../sql-reference/data-types/int-uint.md) or [Array](../../sql-reference/data-types/array.md).
+
+!!! warning "Warning"
+    Currently `Map` data type is an experimental feature. To work with it you must set `allow_experimental_map_type = 1`.
+
+To get the value from an `a Map('key', 'value')` column, use `a['key']` syntax.
```
Comment (CurtizJ) [inline docs/en/sql-reference/data-types/map.md:17]: I think, it's necessary to mention, that lookup works now with a linear complexity.

**Pull request merged**